Improvements

MoneyWasted opened the pull request:
I recently started using this in one of my projects and made some changes to the code to better fit my project's style. I also adjusted how the animation scale is handled, among a few other minor modifications. Feel free to take a look at my changes and offer feedback if needed.

MoneyWasted committed:
Various Improvements & Optimizations

MoneyWasted committed:
Updating README.md

MoneyWasted committed:
Use Current Directory for ImGui Includes

aiekick commented:
Hello,

Thanks for your PR.

But its difficult to see what you changes, since you have done a code style refactor...

MoneyWasted commented:
The main changes are as followed:
* Replaced direct computation in the `if-else` block related to `anim_scale` with a `clamp` function, which limits the value of `anim_scale` between 0 and 1.
* Introduced the `USE_DOCKING` macro and conditionally added window flags if it's set to true and did the same with the already existing `ENABLE_DEBUG`.
* Removed the separate typedef and enum for `ImCoolBarFlags`, and replaced it with an enum inside the ImGui namespace. Also removed prefix `ImCoolBarFlags_` from the enum constants.
* Some extra spaces, line breaks and empty comments were removed for cleaner code.

aiekick commented:
im ok with the two first changes, but not with the others. so i will not merge your PR
